File: BubbleSort.py
# Bubble sort

def bubble_sort(array):
    for i in range(len(array) - 1):
        check = False
        for j in range(len(array) - i - 1):
            if array[j] > array[j+1]:
            #swap
               array[j+1], array[j] = array[j], array[j+1]
               check = True
        if not check:
            return array     
    return array

array = [6, 3, 2, 3, 7, 9, 10]
print(bubble_sort(array))            

File: BubbleSort.c
#include<stdio.h>
#include<stdbool.h>

void bubble_sort(int arr[], int size) {
    // bubble sort
    for(int i = 0; i < size - 1; i++) {
        bool check = false;
        for(int j = 0; j < size - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                // swap
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                check = true;
            }
        }
        if(check == false) {
            break;
        }
    }
}

void printarr(int arr[], int size) {
    
    for(int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
}

int main() {
    int arr[] = {2, 4, 56, 5, 1, 4};
    int size = sizeof(arr) / sizeof(arr[0]);

    bubble_sort(arr, size);
    printarr(arr, size);
    return 0;
}